aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,317 @@
-.. include:: /../common/stub-topic.txt
+:version: 1.0.0
 
-|stub-icon| Overview of ASP.NET MVC
-===================================
+Overview of ASP.NET Core MVC
+============================
 
-.. include:: /../common/stub-notice.txt
+By `Steve Smith`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/113
+ASP.NET Core MVC is a rich framework for building web apps and APIs using the Model-View-Controller design pattern.
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+What is the MVC pattern?
+------------------------
+
+The Model-View-Controller (MVC) architectural pattern separates an application into three main groups of components: Models, Views, and Controllers. This pattern helps to achieve `separation of concerns <http://deviq.com/separation-of-concerns/>`_. Using this pattern, user requests are routed to a Controller which is responsible for working with the Model to perform user actions and/or retrieve results of queries. The Controller chooses the View to display to the user, and provides it with any Model data it requires.
+
+The following diagram shows the three main components and which ones reference the others:
+
+.. image:: overview/_static/mvc.png
+
+This delineation of responsibilities helps you scale the application in terms of complexity because it’s easier to code, debug, and test something (model, view, or controller) that has a single job (and follows the `Single Responsibility Principle <http://deviq.com/single-responsibility-principle/>`_). It's more difficult to update, test, and debug code that has dependencies spread across two or more of these three areas. For example, user interface logic tends to change more frequently than business logic. If presentation code and business logic are combined in a single object, you have to modify an object containing business logic every time you change the user interface. This is likely to introduce errors and require the retesting of all business logic after every minimal user interface change.
+
+.. note:: Both the view and the controller depend on the model. However, the model depends on neither the view nor the controller. This is one the key benefits of the separation. This separation allows the model to be built and tested independent of the visual presentation.
+
+Kinds of Models
```

Comment:
I agree - can we have a doc under Models that talks about this, though? When building real apps that involve more than one project, the docs/samples traditionally have fallen short and they're not leading devs toward building loosely-coupled, maintainable apps. Most devs looking at MVC 1-5 samples/articles would conclude that an EF DbContext *is* the model, which (for all but the simplest hello world apps) isn't a good way to architect any substantial app.

Can I volunteer to write some docs on Models Overview or Model Architecture or something similar, where I could demonstrate these concerns and have the space to show the tradeoffs involved in using them (or not)? Thanks.